nextflow_schema.json

```diff
@@ -0,0 +1,290 @@
+                    "default": "dbscan",
+                    "description": "Cluster contigs using which clustering method. Choices are \"dbscan\" and \"hdbscan\""
+                },
+                "classification_method": {
+                    "type": "string",
+                    "default": "decision_tree",
+                    "description": "Which clustering method to use for unclustered recruitment step. Choices are \"decision_tree\" and \"random_forest\""
+                },
+                "classification_kmer_pca_dimensions": {
+                    "type": "integer",
+                    "default": 50,
+                    "description": "Number of dimensions of which to reduce the initial k-mer frequencies matrix"
+                },
+                "completeness": {
+                    "type": "number",
+                    "default": 20.0,
+                    "description": "Minimum completeness needed to keep a cluster (default is atleast 20% complete)"
+                },
+                "purity": {
+                    "type": "number",
+                    "default": 95.0,
+                    "description": "Minimum purity needed to keep a cluster (default is at least 95% pure)"
+                },
+                "gc_stddev_limit": {
+                    "type": "number",
+                    "default": 5.0,
+                    "description": "Which clusters to keep depending on the GC% std.dev (default is 5%)"
+                },
+                "cov_stddev_limit": {
+                    "type": "number",
+                    "default": 25.0,
+                    "description": "Which clusters to keep depending on the covergae std.dev (default is 25%)"
+                }
+            },
+            "required": [
+                "length_cutoff",
+                "norm_method",
+                "pca_dimensions",
+                "embedding_method",
+                "embedding_dimensions",
+                "kmer_size",
+                "clustering_method",
+                "classification_method",
+                "classification_kmer_pca_dimensions",
+                "completeness",
+                "purity",
+                "gc_stddev_limit",
+                "cov_stddev_limit"
+            ]
+        },
+        "autometa_taxonomy_aware_binning_parameters": {
+            "title": "Autometa Taxonomy-Aware Binning Parameters",
+            "type": "object",
+            "description": "",
+            "default": "",
+            "properties": {
+                "taxonomy_aware": {
+                    "type": "boolean",
+                    "description": "Turn taxonomy-aware clustering on/off"
+                },
+                "single_db_dir": {
+                    "type": "string",
+                    "default": "null",
+                    "fa_icon": "fas fa-folder-open",
+                    "description": "Directory containing databases required for taxonomy assignment"
```

Comment:
Yes. 
There is currently no implementation of hmmpressing the markers so currently everything uses docker because markers is setup correctly in the container.
You had put code for this in another comment I think. But this PR was mostly to reorganize and not implement new/missing functions so I'm going to hold off until this is merged.

Reply:
To clarify, the markers directory is contained inside the docker container so currently it is not needed